osu.Game.Tests/Visual/Menus/TestSceneToolbar.cs

```diff
@@ -21,24 +25,62 @@ public class TestSceneToolbar : OsuTestScene
             typeof(ToolbarNotificationButton),
         };
 
-        public TestSceneToolbar()
+        private Toolbar toolbar;
+
+        [Resolved]
+        private RulesetStore rulesets { get; set; }
+
+        [SetUp]
+        public void SetUp() => Schedule(() =>
         {
-            var toolbar = new Toolbar { State = { Value = Visibility.Visible } };
-            ToolbarNotificationButton notificationButton = null;
+            Child = toolbar = new Toolbar { State = { Value = Visibility.Visible } };
+        });
 
-            AddStep("create toolbar", () =>
-            {
-                Add(toolbar);
-                notificationButton = toolbar.Children.OfType<FillFlowContainer>().Last().Children.OfType<ToolbarNotificationButton>().First();
-            });
+        [Test]
+        public void TestNotificationCounter()
+        {
+            ToolbarNotificationButton notificationButton = null;
 
-            void setNotifications(int count) => AddStep($"set notification count to {count}", () => notificationButton.NotificationCount.Value = count);
+            AddStep("retrieve notification button", () => notificationButton = toolbar.ChildrenOfType<ToolbarNotificationButton>().Single());
 
             setNotifications(1);
             setNotifications(2);
             setNotifications(3);
             setNotifications(0);
             setNotifications(144);
+
+            void setNotifications(int count)
+                => AddStep($"set notification count to {count}",
+                    () => notificationButton.NotificationCount.Value = count);
+        }
+
+        [TestCase(false)]
+        [TestCase(true)]
+        public void TestRulesetSwitchingShortcut(bool toolbarHidden)
+        {
+            ToolbarRulesetSelector rulesetSelector = null;
+
+            if (toolbarHidden)
+                AddStep("hide toolbar", () => toolbar.Hide());
+
+            AddStep("retrieve ruleset selector", () => rulesetSelector = toolbar.ChildrenOfType<ToolbarRulesetSelector>().Single());
+
+            for (int i = 0; i < 4; i++)
```

Comment:
it would've been better to check on the count of available rulesets but I don't think I can